lua/neoclip/telescope.lua

```diff
@@ -92,6 +105,26 @@ local function get_delete_handler(typ)
     end
 end
 
+local function get_edit_handler (typ, telescope_opts)
+    return function(prompt_bufnr)
+        local entry = action_state.get_selected_entry()
+        actions.close(prompt_bufnr)
+        utils.open_editor_temp_window(
+            entry.contents,
+            entry.filetype,
+            function (new_lines)
+                local new_entry = {
+                    contents=new_lines,
+                    filetype=entry.filetype,
+                    regtype=entry.regtype,
+                }
+                storage.replace(typ, entry, new_entry)
+                resume_telescope_at_index(typ, telescope_opts, entry.index)
```

Comment:
Not sure how difficult it would be but this might not open the picker at the edited entry if you did some yanks while editing the entry. I guess we could get the index from the sorted set but that's an O(n) operation.

Reply:
I really hadn't thought of that. I'll take a look in what can be done to solve this problem

Reply:
@AckslD I just changed the implementation to make it work in these situations. I ended up implementing the O(n) operation, but with a few caveats ([commit](https://github.com/AckslD/nvim-neoclip.lua/pull/93/commits/0761680e1fedc51fa1ca50ea5994f3cc64580bce) here)

In my implementation, I also provide an `index_hint`, which is the previous index of the entry. This index can be used as a starting point for the search. Therefore, the number of items searched until a match is the number of entries added to the list (worst case). This would not work if the user managed to remove an item from the list, in which case telescope would simply not pre-select the entry. But I couldn't think of a way for an item to be removed without nuking the database while still editing an entry.

However, I'm not really happy with this implementation. I only wrote this because it was the simplest way to fix the problem. I thought of some ways to improve this, but was not sure they'd be good implementations:
- **Better way to compare entries**: currently the only way to compare two entries is to use `sorted_set.hash` in both, and compare both hashes. This can be expensive, especially if the contents of the entries is large. A way to mitigate this is to add a proper id to the entries. The use of ids could even help if new features require changing the contents of the entry objects
- **Add an event system to `sorted_set`**: There could be a way for the `sorted_set` to let us know when a new entry has been added, and use this to update the index of the entry currently being edited. This seems a bit overengineering for a simple problem, but is a possible solution.

Anyway, these solutions seem a bit too complicated for this PR. I think the current implementation works and is not terrible. So I'd say we could leave it like that, and open another PR fixing this issue, if desired. What do you think?

Reply:
Awesome @Jeansidharta! I think this seems fine also, especially since we'd only have to search through the new entries.

One easy optimization could maybe be to just add the hash to the entry so that it doesn't have to be re-computed every time. Then the hash function could use this if it's there. We would need to make sure it's updated/reset when we edit the entry though.

Anyway, I'm happy with this anyway :)